WMF Framework/CacheItemMigrationPolicy.swift

```diff
@@ -27,25 +27,25 @@ class CacheItemMigrationPolicy: NSEntityMigrationPolicy {
             destinationItem.setValue(newVariant, forKey: "variant")
             destinationItem.setValue(date, forKey: "date")
             
-            //artifically create and save image response header
             var isDownloaded = false
             if let fileName = fetcher.uniqueFileNameForItemKey(key, variant: newVariant),
                 let headerFileName = fetcher.uniqueHeaderFileNameForItemKey(key, variant: newVariant) {
+                
                 let filePath = CacheFileWriterHelper.fileURL(for: fileName).path
+                isDownloaded = FileManager.default.fileExists(atPath: filePath)
                 
-                if let mimeType = FileManager.default.getValueForExtendedFileAttributeNamed(WMFExtendedFileAttributeNameMIMEType, forFileAtPath: filePath),
-                    let data = FileManager.default.contents(atPath: filePath) {
-                        //construct response header file
-                    let headerFields: [String: String] = [
-                            "Content-Type": mimeType,
-                            "Content-Length": String(data.count)
-                        ]
-                    CacheFileWriterHelper.saveResponseHeader(headerFields: headerFields, toNewFileName: headerFileName) { (result) in
-                        switch result {
-                        case .success, .exists:
-                            isDownloaded = true
-                        case .failure:
-                            break
+                //artifically create and save image response header
+                //perform on another thread in case this takes a while
+                DispatchQueue.global(qos: .background).async {
```

Comment:
This could perform too many file operations concurrently since there's not a good built-in concurrency limit on dispatch queues. Also, the db migration itself happening on the main thread could block long enough to cause the system to kill the app. I committed an update that moves the whole migration to a background queue, but feel free to revert if I'm missing something.

Reply:
@joewalsh thanks for catching these! Changes look good to me 👍